src/main/java/org/spongepowered/common/data/DataRegistrar.java

```diff
@@ -586,6 +586,14 @@ public static void setupSerialization(Game game) {
         dataManager.registerDualProcessor(DirectionalData.class, SpongeDirectionalData.class, ImmutableDirectionalData.class,
                 ImmutableSpongeDirectionalData.class, new SkullRotationDataProcessor());
 
+        final AchievementDataProcessor achievementDataProcessor = new AchievementDataProcessor();
+        dataManager.registerDataProcessorAndImpl(AchievementData.class, SpongeAchievementData.class, ImmutableAchievementData.class,
```

Comment:
You should now use `registerDualProcessor` for `AchievementDataProcessor` and `StatisticDataProcessor` since it's a single key. Make the processors subclasses of `AbstractSingleDataSingleTargetProcessor` (or one of it's derivatives)